backend/src/main/java/ai/verta/modeldb/utils/RdbmsUtils.java

```diff
@@ -161,9 +170,18 @@ public static KeyValueEntity generateKeyValueEntity(
     LOGGER.trace("Converting KeyValueEntityListFromKeyValues ");
     List<KeyValue> attributeList = new ArrayList<>();
     if (keyValueEntityList != null) {
-      for (KeyValueEntity keyValue : keyValueEntityList) {
-        attributeList.add(keyValue.getProtoKeyValue());
-      }
+      attributeList =
+          keyValueEntityList.stream()
+              .map(
+                  keyValueEntity -> {
+                    try {
+                      return keyValueEntity.getProtoKeyValue();
+                    } catch (InvalidProtocolBufferException e) {
+                      throw new RuntimeException(e);
+                    }
```

Comment:
Please extract to private method. This will collapse the size of the stream processing block and make the code easier to read.

Reply:
extracted